### PR TITLE
Revamped refpix to be more efficient

### PR DESCRIPTION
**Pull request by stsci-hack**
This version of refpix.reference_pixels completely revises the memory usage of the algorithm, while also resulting in improved run-times (from 3m16s to 23s for 250,12,256,2048 NRCALONG simulated observation).

### CODE REVIEW

File: jwst/refpix/reference_pixels.py [hunk 1]
```diff
@@ -788,279 +823,300 @@ def do_subarray_corrections(self):
         #
         #  First transform to detector coordinates
         #
-        self.DMS_to_detector()
         refdq = dqflags.pixel['REFERENCE_PIXEL']
```
Comment (stscirij) [inline jwst/refpix/reference_pixels.py]: need to add
self.DMS_to_detector(0,0)
to make sure the data and dq groups are shifted/rotated to DMS coordinates

Comment (stsci-hack) [reply jwst/refpix/reference_pixels.py]: Addressed with commit stsci-hack/jwst@47dd1d9b839fff2d14806580c18a5a1ed3d72da4.

Comment (stsci-hack) [reply jwst/refpix/reference_pixels.py]: This should now be addressed properly with 86be1ed.

Comment (stscirij) [reply jwst/refpix/reference_pixels.py]: This looks good now

**Pull request merged**